state-chain/pallets/cf-swapping/src/tests/fees.rs

```diff
@@ -127,45 +127,97 @@ fn test_buy_back_flip() {
 
 #[test]
 fn test_network_fee_calculation() {
```

Comment:
A couple of things I think we can improve on in this test:
1. Test the `CollectedNetworkFee` storage is as we expect - this is possibly the most important and it's not tested here.
2. Rather than insert a swap request over the top of another request, why not just have the function take an optional swap request id rather than this `with_swap_request: bool` ?  Then the same id can be used for multiple tests and it's clear that the intention is that it's the same swap request, rather than each invocation actually overwriting the swap request, making it look like each one is new (which it actually kinda is at the moment).